core/domain/opportunity_services.py

```diff
@@ -498,27 +463,50 @@ def get_translation_opportunities(language_code, topic_name, cursor):
         .ExplorationOpportunitySummaryModel.get_all_translation_opportunities(
             page_size, cursor, language_code, topic_name))
     opportunities = []
-    suggestion_ids = []
-    opportunity_ids = []
+    exp_id_to_in_review_count = (
+        _build_exp_id_to_translation_suggestion_in_review_count(
+            exp_opportunity_summary_models, language_code))
+    for exp_opportunity_summary_model in exp_opportunity_summary_models:
+        exp_opportunity_summary = (
+            get_exploration_opportunity_summary_from_model(
+                exp_opportunity_summary_model))
+        if exp_opportunity_summary.id in exp_id_to_in_review_count:
+            exp_opportunity_summary.translation_in_review_counts = {
```

Comment:
I'm a bit confused by this -- it looks as though we're modifying the domain objects because the information there is out of date. Is that right? If so then do we persist these changes subsequently?

Maybe adding a comment would be useful.

Reply:
It's a good question -- I think that either way could work. I think the main heuristic is to try and minimize the part of the code where the domain object is in a "not yet complete" state -- so ideally the domain object gets formed and fully populated within a single function.

For now, maybe let's keep it as-is but add a comment in the domain object to explain what's going on with that field (something like "This field needs to be separately populated" or similar)?

The alternative would be to completely separate that field from the domain object, and have the backend controller fetch that info separately for assembly into the response sent to the frontend. That actually feels cleaner to me, and should be feasible since this is a top-level property, but would it increase the time/query complexity in this case?